test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -1106,6 +1111,27 @@ Resource not found
                 """);
     }
 
+    /// <summary>
+    /// Scripts in repo root should not include <c>.resx</c> files.
+    /// Part of <see href="https://github.com/dotnet/sdk/issues/49826"/>.
+    /// </summary>
+    [Fact]
+    public void EmbeddedResource_AlongsideSln()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory();
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.cs"), s_programReadingEmbeddedResource);
+        File.WriteAllText(Path.Join(testInstance.Path, "Resources.resx"), s_resx);
+        File.WriteAllText(Path.Join(testInstance.Path, "repo.sln"), "");
```

Comment:
Consider also including cases for the other file types `.csproj` `.slnx` we have this behavior for, as well as a file type where we don't exclude default items but may consider doing so in future, such as `.shproj`.